From a8309eb056d7763ad2b5f188df3cb64b5eaa2132 Mon Sep 17 00:00:00 2001
From: Patrik Flykt <patrik.flykt@linux.intel.com>
Date: Fri, 2 Dec 2011 13:55:37 +0200
Subject: [PATCH] service: Disconnect network when service is freed

Call __connman_network_disconnect() when a service is freed.
This prevents wpa_supplicant from trying to connect forever when
ConnMan is shut down.
---
 src/service.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/service.c b/src/service.c
index fe4b51b..4111070 100644
--- a/src/service.c
+++ b/src/service.c
@@ -3464,6 +3464,7 @@ static void service_free(gpointer user_data)
 	g_hash_table_destroy(service->counter_table);
 
 	if (service->network != NULL) {
+		__connman_network_disconnect(service->network);
 		if (service->network_created == TRUE)
 			connman_network_unref(service->network);
 	}
-- 
1.7.9

